src/bvals/comms/bnd_info.cpp

```diff
@@ -77,35 +82,49 @@ SpatiallyMaskedIndexer6D CalcIndices(const NeighborBlock &nb,
                                      TopologicalElement el, IndexRangeType ir_type,
                                      bool prores, std::array<int, 3> tensor_shape) {
   const auto &ni = nb.ni;
-  auto loc = pmb->loc;
+  const auto &loc = pmb->loc;
   auto shape = pmb->cellbounds;
   // Both prolongation and restriction always operate in the coarse
   // index space. Also need to use the coarse index space if the
   // neighbor is coarser than you, wether or not you are setting
   // interior or exterior cells
   if (prores || nb.loc.level() < loc.level()) shape = pmb->c_cellbounds;
 
+  // Re-create the index space for the neighbor block (either the main block or
+  // the coarse buffer as required)
+  int coarse_fac = 1;
+  if (nb.loc.level() > loc.level()) coarse_fac = 2;
+  auto neighbor_shape = IndexShape(nb.block_size.nx(X3DIR) / coarse_fac,
+                                   nb.block_size.nx(X2DIR) / coarse_fac,
+                                   nb.block_size.nx(X1DIR) / coarse_fac, Globals::nghost);
+
   IndexDomain interior = IndexDomain::interior;
   std::array<IndexRange, 3> bounds{shape.GetBoundsI(interior, el),
                                    shape.GetBoundsJ(interior, el),
                                    shape.GetBoundsK(interior, el)};
+  std::array<IndexRange, 3> neighbor_bounds{neighbor_shape.GetBoundsI(interior, el),
+                                            neighbor_shape.GetBoundsJ(interior, el),
+                                            neighbor_shape.GetBoundsK(interior, el)};
 
+  std::array<bool, 3> not_symmetry{!pmb->block_size.symmetry(X1DIR),
+                                   !pmb->block_size.symmetry(X2DIR),
+                                   !pmb->block_size.symmetry(X3DIR)};
```

Comment:
`symmetry` implies that this is a direction that possesses translational symmetry (or for other coordinate systems possibly rotational symmetry), so that we don't need to include ghost zones. Previously we assumed if a direction had size one, then it was a symmetry direction (or thought of differently we reduced the dimensionality of the problem). The problem now is that as we reduce blocks sizes below the root grid, we can end up with directions that are size one that aren't symmetry directions.

Reply:
Agreed. The check for `nx# == 1`is not the same thing.
Might be worth discussing to introduce sth (global) like `ndim` or `has_ghosts<XNDIR>()`

Reply:
Certainly we could switch to something like that, but I think that might be for another PR.